src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/CSharp/BindTagHelperDescriptorProvider.cs

```diff
@@ -343,7 +343,8 @@ protected override void Collect(ISymbol symbol, ICollection<TagHelperDescriptor>
             private (string Type, string Namespace)? _names;
 
             public (string Type, string Namespace) GetNames()
-                => _names ??= (_type.ToDisplayString(), _type.ContainingNamespace.ToDisplayString());
+                => _names ??= (_type.ToDisplayString(),
+                    _type.ContainingNamespace.ToDisplayString(SymbolExtensions.FullNameTypeDisplayFormat));
```

Comment:
`_type.ContainingNamespace.ToDisplayString()` would be `"<global namespace>"`; with `SymbolExtensions.FullNameTypeDisplayFormat` we get an empty string instead (this format was already used in other places, just not everywhere)